app/models/user.rb

```diff
@@ -68,6 +68,11 @@ def self.from_omniauth(auth)
       actor.email = actor.email.presence || auth.info.email
       actor.given_name = actor.given_name.presence || auth.info.given_name
       actor.surname = actor.surname.presence || auth.info.surname || auth.info.family_name
+      begin
+        actor.orcid = PsuIdentity::DirectoryService::Client.new.userid(auth.uid).orc_id
+      rescue StandardError => e
+        logger.error(e)
+      end
```

Comment:
I agree with Nick, let's move this to its own method

Reply:
I would also feel more comfortable if we did some amount of parsing or formatting on the orcid that PSU Identity returns back. It's possible that PSU Identity could return us back either a malformed orcid, or an orcid in a different format than the canonical way that we store them. Because we have logic that looks up orcids in the db, we need to ensure that they are all stored in the same format as the lookup services are using.

Reply:
It seems like the safest thing to do would be to Run the id though `OrcidId(orcid).to_s` and store that value, then `.uri` for display?

Reply:
reminder that orcids are not just digits.

Reply:
"X" is a digit in Roman numerals